rts/Game/Action.h

```diff
@@ -5,21 +5,35 @@
 
 #include <string>
 #include "Game/UI/KeySet.h"
-#include "Game/UI/KeySetSC.h"
 
 class Action
 {
 public:
 	Action() {}
 	Action(const std::string& line);
 
-	std::string command;   ///< first word, lowercase
-	std::string extra;     ///< everything but the first word
-	std::string rawline;   ///< includes the command, case preserved
-	std::string boundWith; ///< the string that defined the binding keyset
+	int         bindingIndex; ///< the order for the action trigger
+	std::string command;      ///< first word, lowercase
+	std::string extra;        ///< everything but the first word
+	std::string rawline;      ///< includes the command, case preserved
+	std::string boundWith;    ///< the string that defined the binding keyset
+	CKeyChain   keyChain;     ///< the bound keychain/keyset
 
-	CKeyChain keyChain;    ///< the bounded keychain/keyset
-	CKeyChainSC keyChainSC;
+	bool operator<(const Action& action) const
+	{
+		bool selfAnyMod = keyChain.back().AnyMod();
+		bool actionAnyMod = action.keyChain.back().AnyMod();
+
+		if (selfAnyMod == actionAnyMod)
+			return bindingIndex < action.bindingIndex;
+		else
+			return actionAnyMod;
+	}
+
+	bool operator==(const Action& action) const
+	{
+		return command == action.command && extra == action.extra;
+	}
```

Comment:
I assume that these two comparison functions are used for sorting & the unique thing. However, since they are actually doing different things (the '<' checks for the binding index, while '==' checks for the command), if you are only using them in a single place I'd probably prefer them to be defined as lambdas at the point of use.

Reply:
Alternatively, if you use them in more than one place, you can even just give them another name (like orderByIndex and compareByAction) and then still pass the functions to the algorithms.

Reply:
Since you're looking at this, can you tell me if std::unique only works if the duplicates are adjacent?

Reply:
Yeah, so you always need to sort first.

Reply:
Yeah, deduplication won't work in the current implementation.

Reply:
This is resolved as of the last commit

Reply:
Not really, I actually had some trouble extracting the comparison functions. I intend to proceed as is for now and we can then iterate on a refactor

Reply:
Ok, done.